src/ronin/basefee-treasury/RoninBaseFeeTreasury.sol

```diff
@@ -0,0 +1,399 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.27;
+
+import { Initializable } from "@openzeppelin-v5/contracts/proxy/utils/Initializable.sol";
+import { EIP712 } from "@openzeppelin-v5/contracts/utils/cryptography/EIP712.sol";
+import { SafeCast } from "@openzeppelin-v5/contracts/utils/math/SafeCast.sol";
+import { EnumerableSet } from "@openzeppelin-v5/contracts/utils/structs/EnumerableSet.sol";
+
+import { HasContracts } from "src/extensions/collections/HasContracts.sol";
+import { GlobalConfigConsumer } from "src/extensions/consumers/GlobalConfigConsumer.sol";
+
+import { IProfile } from "src/interfaces/IProfile.sol";
+import { IStaking } from "src/interfaces/staking/IStaking.sol";
+import { ICandidateManager } from "src/interfaces/validator/ICandidateManager.sol";
+
+import { IBaseFeeTreasury } from "src/interfaces/basefee-treasury/IBaseFeeTreasury.sol";
+
+import { ContractType } from "src/utils/ContractType.sol";
+
+import { ErrEmptyArray, ErrLengthMismatch, ErrUnauthorized, RoleAccess } from "src/utils/CommonErrors.sol";
+
+import { ErrorHandler } from "src/libraries/ErrorHandler.sol";
+import { LibArray } from "src/libraries/LibArray.sol";
+
+contract RoninBaseFeeTreasury is EIP712, Initializable, HasContracts, GlobalConfigConsumer, IBaseFeeTreasury {
+  using LibArray for *;
+  using SafeCast for uint256;
+  using ErrorHandler for bool;
+  using EnumerableSet for EnumerableSet.Bytes32Set;
+
+  uint256 public constant COOLDOWN_PERIOD = 1 days;
+  uint256 public constant MIN_PROPOSAL_DURATION = 1 days;
+  bytes32 internal constant _TYPE_HASH = keccak256(
+    "Proposal(address proposer,uint256 nonce,uint256 expiry,address executor,address[] recipients,uint256[] amounts,bytes[] callDatas)"
+  );
```

Comment:
Make sure the data types are matched with the proposal's structure

Reply:
will add to unit test

Reply:
added.